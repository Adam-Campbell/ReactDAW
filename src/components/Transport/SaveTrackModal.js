import React from 'react';
import PropTypes from 'prop-types';

const SaveTrackModal = props => (
    <div className="modal__overlay">
        <div className="modal__content-container">
            <button
                className="button pink"
                onClick={props.handleClose}
            >Close</button>
            <div>
                <input
                    className="modal__text-input"
                    type="text"
                    value={props.saveName}
                    onChange={props.updateSaveName}
                    placeholder="Choose a name"
                ></input>
                <button 
                    className="button pink"
                    onClick={props.handleSave}
                >Save Track</button>
            </div>
        </div>
    </div>
);

SaveTrackModal.propTypes = {
    handleClose: PropTypes.func.isRequired,
    saveName: PropTypes.string.isRequired,
    updateSaveName: PropTypes.func.isRequired,
    handleSave: PropTypes.func.isRequired
};

export default SaveTrackModal;