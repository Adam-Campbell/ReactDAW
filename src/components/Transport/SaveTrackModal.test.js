import React from 'react';
import { shallow } from 'enzyme';
import SaveTrackModal from './SaveTrackModal';

test('renders correctly', () => {
    const component = shallow(
        <SaveTrackModal 
            handleClose={jest.fn()}
            handleSave={jest.fn()}
            updateSaveName={jest.fn()}
            saveName="New Track"
        />
    );
    expect(component).toMatchSnapshot();
});
