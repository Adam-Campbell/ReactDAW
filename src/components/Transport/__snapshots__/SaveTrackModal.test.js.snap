// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="modal__overlay"
>
  <div
    className="modal__content-container"
  >
    <button
      className="button pink"
      onClick={[MockFunction]}
    >
      Close
    </button>
    <div>
      <input
        className="modal__text-input"
        onChange={[MockFunction]}
        placeholder="Choose a name"
        type="text"
        value="New Track"
      />
      <button
        className="button pink"
        onClick={[MockFunction]}
      >
        Save Track
      </button>
    </div>
  </div>
</div>
`;
